test/integration/api/users.js

```diff
@@ -74,6 +77,38 @@ describe('api: /users', () => {
             body[0].email.should.equal('alice@getodk.org');
             body[0].displayName.should.equal('Alice');
           }))));
+
+    it('should return lastLoginAt as null for users who have never logged in', testService((service) =>
+      service.login('alice', (asAlice) =>
+        asAlice.post('/v1/users')
+          .send({ email: 'newuser@getodk.org' })
+          .expect(200)
+          .then(() => asAlice.get('/v1/users/?q=newuser@getodk.org')
+            .expect(200)
+            .then(({ body }) => {
+              body.length.should.equal(1);
+              body[0].email.should.equal('newuser@getodk.org');
+              should(body[0].lastLoginAt).be.null();
+            })))));
+
+    it('should update lastLoginAt when user logs in multiple times', testService(async (service) => {
+      const asAlice = await service.login('alice');
+
+      const firstLogin = await asAlice.get('/v1/users/current')
+        .expect(200)
+        .then(({ body }) => body);
+      should(firstLogin.lastLoginAt).not.be.null();
+      const firstLoginTime = new Date(firstLogin.lastLoginAt);
+
+      await service.login('alice');
+      const secondLogin = await asAlice.get('/v1/users/current')
+        .expect(200)
+        .then(({ body }) => body);
+      should(secondLogin.lastLoginAt).not.be.null();
+      const secondLoginTime = new Date(secondLogin.lastLoginAt);
+
+      secondLoginTime.should.be.greaterThan(firstLoginTime);
```

Comment:
I feel like there's a little bit of a risk of an intermittent test failure here. If the test ran within a millisecond, this assertion would fail. You could throw in something to wait for a millisecond. Or we could leave it and see if it's a problem in practice.